flax/linen/attention.py

```diff
@@ -202,6 +202,54 @@ def dot_product_attention(
 class MultiHeadDotProductAttention(Module):
   """Multi-head dot-product attention.
 
+  Example usage::
+
+    >>> import flax.linen as nn
+    >>> import jax
+
+    >>> layer = nn.MultiHeadDotProductAttention(num_heads=8, qkv_features=16)
+    >>> key1, key2, key3, key4, key5, key6 = jax.random.split(jax.random.key(0), 6)
+    >>> shape = (4, 3, 2, 5)
+    >>> q, k, v = jax.random.uniform(key1, shape), jax.random.uniform(key2, shape), jax.random.uniform(key3, shape)
+    >>> variables = layer.init(jax.random.key(0), q)
+
+    >>> # equivalent to layer.apply(variables, inputs_q=q, inputs_k=q) and layer.apply(variables, inputs_q=q, inputs_k=q, inputs_v=q)
```

Comment:
maybe reverse the order? show the full signature first `out = layer.apply(variables, q, k, v)` and end with `out = layer.apply(variables, q)`